pkg/csi/cinder/nodeserver.go

```diff
@@ -180,6 +295,42 @@ func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	//TODO: ephemeralVolume := req.GetVolumeContext()["csi.storage.k8s.io/ephemeral"] == "true"
+	if mode == "ephemeral" {
+		return nodeUnpublishEphermeral(req, ns, &vol)
```

Comment:
cant we move above code into noeUnpublishEphemeral?

Reply:
is it to avoid code duplicacy  for unmount operation, i kept it here